firestore/src/common/settings.cc

```diff
@@ -70,6 +121,40 @@ std::ostream& operator<<(std::ostream& out, const Settings& settings) {
   return out << settings.ToString();
 }
 
+bool operator==(const Settings& lhs, const Settings& rhs) {
+  if (lhs.host() != rhs.host()) {
+    return false;
+  }
+
+  if (lhs.is_ssl_enabled() != rhs.is_ssl_enabled()) {
+    return false;
+  }
+
+  if (lhs.cache_settings_source_ != rhs.cache_settings_source_) {
+    return false;
+  }
+
+  if (*lhs.local_cache_settings_ != *rhs.local_cache_settings_) {
```

Comment:
What happens if `lhs` and `rhs` have different types of persistence? For example, what if `*lhs.local_cache_settings_` is memory local cache and `*rhs.local_cache_settings_` is persistent local cache? Which `operator==` is used to compare them? I'm actually surprised this compiles, and doesn't crash if they are different types.